iphone/Maps/UI/Editor/MWMEditorViewController.mm

```diff
@@ -724,6 +724,19 @@ - (void)fillCell:(UITableViewCell * _Nonnull)cell atIndexPath:(NSIndexPath * _No
   }
 }
 
+- (void)removeAdditionalNameAtIndexPath:(NSIndexPath *)indexPath
+{
+  MWMEditorAdditionalNameTableViewCell * cell = [self.tableView cellForRowAtIndexPath:indexPath];
+  if (!cell)
+    NSAssert(false, @"Removed MWMEditorAdditionalNameTableViewCell should not be nil");
+  m_mapObject.RemoveNameForLanguage(cell.code);
```

Comment:
@biodranik 
When we type some text into the additional name field this data will be saved immediately while typing:
```objc
- (void)cell:(MWMTableViewCell *)cell changedText:(NSString *)changeText
{
  NSAssert(changeText != nil, @"String can't be nil!");
  NSIndexPath * indexPath = [self.tableView indexPathForRowAtPoint:cell.center];
  MWMEditorCellID const cellType = [self cellTypeForIndexPath:indexPath];
  std::string val = changeText.UTF8String;
  BOOL isFieldValid = YES;
  switch (cellType)
  {
  case MWMEditorCellTypeBuilding:
    m_mapObject.SetHouseNumber(val);
    isFieldValid = osm::EditableMapObject::ValidateHouseNumber(val);
    break;
  case MWMEditorCellTypeAdditionalName:
    isFieldValid = osm::EditableMapObject::ValidateName(val);
    if (isFieldValid)
      m_mapObject.SetName(val, static_cast<MWMEditorAdditionalNameTableViewCell *>(cell).code); // <--- saving
    break;
  default:
    auto const metadataID = static_cast<MetadataID>(cellType);
    ASSERT_LESS(metadataID, MetadataID::FMD_COUNT, ());
    isFieldValid = osm::EditableMapObject::IsValidMetadata(metadataID, val)? YES : NO;
    m_mapObject.SetMetadata(metadataID, std::move(val));
    break;
  }
```

So we should clean up somehow this text while deleting. If I just remove the cell it will note affect the data.

Reply:
@kirylkaveryn is it correct to save data while typing instead of saving it when a "save" button is pressed?